file-events/src/file-events/cpp/linux_fsnotifier.cpp

```diff
@@ -312,13 +311,14 @@ bool Server::unregisterPath(const u16string& path) {
         return false;
     }
     auto& watchPoint = it->second;
+    int wd = watchPoint.watchDescriptor;
     CancelResult ret = watchPoint.cancel();
     if (ret == CancelResult::ALREADY_CANCELLED) {
         return false;
     }
-    recentlyUnregisteredWatchRoots.emplace(watchPoint.watchDescriptor, path);
-    watchRoots.erase(watchPoint.watchDescriptor);
-    watchPoints.erase(it);
+    recentlyUnregisteredWatchRoots.emplace(wd, path);
```

Comment:
I think we should add a comment here why we don't use the value of `watchPoint.watchDescriptor` any more.

Reply:
I wanted to add that comment about how `cancel()` might already trigger the event, but it doesn't check out. This is basically defensive coding at this point, but I don't have a good rationalization why it's needed.

Reply:
Yeah, adding a comment about being defensive here because if we aren't we are going to get a crash on Alpine would be good enough, even if we don't understand the reason. To make sure we don't accidentally re-introduce the problem when we refactor the code.

Reply:
This is not the cause of the crash; it was caused by the iterator going missing. I added a comment here: https://github.com/gradle/native-platform/pull/282